snow/engine/snowman/transitive.go

```diff
@@ -515,22 +515,26 @@ func (t *Transitive) Start(ctx context.Context, startReqID uint32) error {
 	if oracleBlk, ok := lastAccepted.(snowman.OracleBlock); ok {
 		options, err := oracleBlk.Options(ctx)
 		switch {
-		case err == snowman.ErrNotOracle:
-			// if there aren't blocks we need to deliver on startup, we need to set
-			// the preference to the last accepted block
-			if err := t.VM.SetPreference(ctx, lastAcceptedID); err != nil {
-				return err
-			}
-		case err != nil:
-			return err
-		default:
+		case err == nil:
 			issuedMetric := t.metrics.issued.WithLabelValues(builtSource)
 			for _, blk := range options {
 				// note that deliver will set the VM's preference
 				if err := t.deliver(ctx, t.Ctx.NodeID, blk, false, issuedMetric); err != nil {
 					return err
 				}
 			}
+		case err != snowman.ErrNotOracle:
+			t.Ctx.Log.Warn("failed to set preferences for oracle block",
+				zap.Stringer("blkID", lastAcceptedID),
+				zap.Error(err),
+			)
+			fallthrough
```

Comment:
This would actually the first time we used a `fallthrough` in the repo

Reply:
Why would we not return on an unexpected `err`?

Reply:
returning an error here FATALs the consensus engine. We may be able to still return an error here (as the call to `Options()` should always succeed if the proposal block passed verification...) But I felt like it would be better to continue on and log the error.

Reply:
Is there a path where recovery is possible by proceeding from here? (I'd assume so)

Reply:
If we learn about either the `Commit` or `Abort` block from a peer - then we will add them to consensus like normal. We just won't have initially colored ourselves (so we would be adopting the opinion of the first choice we saw rather than a choice based on our local preference)